workers/waitgroup.go

```diff
@@ -0,0 +1,49 @@
+package workers
+
+import (
+	"sync"
+)
+
+type WaitGroup interface {
+	Add(int)
+	Done()
+	Wait()
+}
+
+type waitGroup struct {
+	count int64
+	mu    *sync.RWMutex
+	cond  *sync.Cond
+}
+
+func (wg *waitGroup) Add(delta int) {
+	wg.mu.Lock()
+	wg.count += int64(delta)
+	if wg.count == 0 {
+		wg.cond.Broadcast()
+	} else if wg.count < 0 {
+		wg.mu.Unlock()
+		panic("waitgroup: negative WaitGroup counter")
```

Comment:
are you sure that you should panic inside the package? this is bad practice

Reply:
see https://github.com/golang/go/blob/master/src/sync/waitgroup.go#L74